src/itwinai/torch/trainer.py

```diff
@@ -1220,3 +1231,390 @@ def dist_train(
+
+        Returns:
+            Tuple[Dataset, Dataset, Dataset, Any]:
+                A tuple containing the datasets and the training result grid.
+        """
+        train_with_data = tune.with_parameters(
+            self.train, data=[train_dataset, validation_dataset, test_dataset]
+        )
+        trainer = ray.train.torch.TorchTrainer(
+            train_with_data,
+            train_loop_config=self.train_loop_config,
+            scaling_config=self.scaling_config,
+            run_config=self.run_config,
+        )
+        param_space = {"train_loop_config": self.train_loop_config}
+        tuner = tune.Tuner(trainer, param_space=param_space, tune_config=self.tune_config)
+
+        result_grid = tuner.fit()
+
+        return train_dataset, validation_dataset, test_dataset, result_grid
+
+    def set_epoch(self, epoch: int) -> None:
+        self.train_dataloader.sampler.set_epoch(epoch)
+        if self.validation_dataloader is not None:
+            self.validation_dataloader.sampler.set_epoch(epoch)
+        if self.test_dataloader is not None:
+            self.test_dataloader.sampler.set_epoch(epoch)
+
+    def _set_tune_config(self):
+        tune_config = self.config.get("tune_config")
+        if tune_config:
+            num_samples = tune_config.get("num_samples", 10)
+            metric = tune_config.get("metric", "loss")
+            mode = tune_config.get("mode", "min")
+
+            search_alg = (
+                get_raytune_search_alg(tune_config) if "search_alg" in tune_config else None
+            )
+            scheduler = (
+                get_raytune_schedule(tune_config) if "scheduler" in tune_config else None
+            )
+
+            # Only set metric and mode if search_alg and scheduler aren't defined
+            self.tune_config = tune.TuneConfig(
+                num_samples=num_samples,
+                metric=metric if not search_alg and not scheduler else None,
+                mode=mode if not search_alg and not scheduler else None,
+                search_alg=search_alg,
+                scheduler=scheduler,
+            )
+        else:
+            print("INFO: No Tune Config configured.")
+            self.tune_config = None
+
+    def _set_scaling_config(self):
+        scaling_config = self.config.get("scaling_config")
+        if scaling_config:
+            self.scaling_config = ray.train.ScalingConfig(**scaling_config)
+        else:
+            print("INFO: No Scaling Config configured.")
+            self.scaling_config = None
+
+    def _set_run_config(self):
+        run_config = self.config.get("run_config")
+        if run_config:
+            storage_path = Path(run_config.get("storage_path"))
+            if storage_path:
+                self.run_config = ray.train.RunConfig(storage_path=Path.absolute(storage_path))
+        else:
+            print("INFO: No RunConfig provided. Assuming local or single-node execution.")
+            self.run_config = None
```

Comment:
In this case it seems to me that you are checking if `run_config` is in the config and that you are only setting `self.run_config` if `storage_path` is a field of it. I had some trouble understanding what was going on here for a second and thus I think I would prefer something like: 

```python
if not "run_config" in self.config: 
    self.run_config = None
    print(...)
    return

run_config = self.config["run_config"]
if not "storage_path" in run_config: 
   return

self.run_config = ray.train...
```